runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -182,7 +184,11 @@ class RTDevice {
         // Besides, this provides support for runtime device (RTD) libraries added to the system
         // path. This maintains backward compatibility for specifying a device using its name.
         // TODO: This support may need to be removed after updating the C++ unit tests.
```

Comment:
Do you see a way of doing away with these special cases somehow? It seems they only exist for testing.

Reply:
Was just curious since you are probably more familiar with C++ testing than us, but feel free to resolve if you don't have any ideas here.

Reply:
Yes, this is a nice question. I thought about that but I have no idea.